src/dbnode/persist/fs/seek_manager.go

```diff
@@ -463,6 +478,9 @@ func (m *seekerManager) Close() error {
 		return errSeekerManagerAlreadyClosed
 	}
 
+	// Unregister for lease updates since all the seekers are going to be closed.
+	m.blockRetrieverOpts.BlockLeaseManager().UnregisterLeaser(m)
```

Comment:
This could potentially cause a deadlock I believe, if block lease manager is calling to the seeker manager to "UpdateLease" and it acquires it's own lock, then it tries to grab the seeker manager lock, but the seeker manager already grabbed it's lock for "Close()", then the seeker manager calls "UnregisterLeaser()" which will try to grab the lease manager lock, but it's already held by a "UpdateLease" call that's waiting for the seeker manager lease to be released.

Maybe just do this outside of the lock in some way?

Reply:
This could deadlock in the "Open" too I believe.